client/component/SelectionConfig.tsx

```diff
@@ -72,7 +80,7 @@ export function SelectionConfig(props: SelectionConfigProps) {
   }
 
   function updateXLength(l: number) {
-    if (currentSelection && 'lengths' in currentSelection) {
+    if (currentSelection && RectangularSelection.isShape(currentSelection)) {
```

Comment:
Refactor this so each selection type can contributes specific UI to config dialogue.